cut-the-crap: init at 1.0.0

jappeace opened the pull request:
###### Motivation for this change

Adds a useful automated video editing tool for streamers.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

I think I'm the maintainer of this package

cc @

vbgl committed:
[Don’t merge] Recurse into OCaml packages

cdepillabout commented:
@jappeace Would you consider releasing this to Hackage?

That way, it would get updates automatically here in nixpkgs whenever you released a new version.

jappeace commented:
Hi @cdepillabout I uploaded it to [hackage](http://hackage.haskell.org/package/cut-the-crap-1.0.0).
It still needs ffmpeg as a runtime dependency, how does one add that?

cdepillabout commented:
@jappeace Thanks!

Next time the hackage packages are updated on the `haskell-updates` branch, you should be able to see `cut-the-crap`.  You'll be able to try to build it with a command like `nix-build -A haskellPackages.cut-the-crap`.

> It still needs ffmpeg as a runtime dependency, how does one add that?

You should be able to add that with a wrapper script.  Here's an example:

https://github.com/NixOS/nixpkgs/blob/a8144b1ad33547cfdc31a903f8f7c0c658d78216/pkgs/development/haskell-modules/configuration-nix.nix#L585-L593

For cut-the-crap, it might look something like this:

```nix
  cut-the-crap =
    let path = stdenv.lib.makeBinPath [ ffmpeg ];
    in overrideCabal (addBuildTool super.futhark makeWrapper) (_drv: {
      postInstall = ''
        wrapProgram $out/bin/cut-the-crap \
          --prefix PATH : "${path}"
      '';
    });

Commit:
airsonic: enable nginx.recommendedProxySettings with virtualHost

This fixes music playback when using the `services.airsonic.virtualHost`
option.

oxalica committed:
rust-analyzer: init at unstable-2020-03-09

oxalica committed:
maintainers: add oxalica

oxalica committed:
vscode-extensions.rust-analyzer: init at unstable-2020-03-09

doronbehar commented on pkgs/applications/video/cut-the-crap/default.nix:
```diff
@@ -0,0 +1,42 @@
+    system-filepath temporary text unliftio-core
+  ];
+  prePatch = "hpack";
+  license = stdenv.lib.licenses.mit;
```
Why is there no `meta` as with any regular derivation? Maybe I'm just not familiar with the Haskell ecosystem...

jappeace replied:
This is generated, but at the moment I'm waiting for haskellPackages to be updated so I can just do cdepillabout's suggestion.

doronbehar commented on pkgs/applications/video/cut-the-crap/default.nix:
```diff
@@ -0,0 +1,42 @@
+  src = fetchgit {
+    url = "https://github.com/jappeace/cut-the-crap";
+    sha256 = "1z4s0zan421q4722k32p0n33bsh7di1m6pbbdzqi7j87lfd9i7nk";
+    rev = "0ea8a0a0fd3e8d85f44337f13ef2445d22d1bf97";
```
Why not use https://github.com/jappeace/cut-the-crap/releases/tag/1.0.0 ?

jappeace replied:
I made this before that debian thing. I wouldn't know how to use tags, or do you just mean the same commit as that tag?

doronbehar replied:
Maybe that's also automatically generated...

offlinehacker committed:
modules/virtualisation: add hyperv-image

This module implements builder for hyper-v images.

Ma27 committed:
nixos/captive-browser: set chromium's data-dir to a XDG-compliant location

To quote the XDG specification:

     There is a single base directory relative to which user-specific
     data files should be written. This directory is defined by the\
     environment variable $XDG_DATA_HOME.

Rather than adding another directory to $HOME, I think that it's better
to follow this standard to avoid a cluttered home-dir.

oxalica committed:
rust-analyzer: 2020-03-09 -> 2020-03-16
vscode-extensions.rust-analyzer: 2020-03-09 -> 2020-03-16

oxalica committed:
rust-analyzer: check compatibility in update script

andir committed:
buildRustCrate: support proc-macro in default prelude

Commit:
ivan: add full iconset

bachp committed:
nixos/boinc: log to journal instead of log file

bachp committed:
nixos/boinc: create boinc group

This allows users that are members of the boinc group
to interact with the boinc service by running:

boincmgr -d /var/lib/boinc

puzzlewolf committed:
line-awesome: init at 1.3.0

danielfullmer committed:
hcxdumptool: init at 6.0.5

anna328p committed:
multimc: 0.6.7 -> 0.6.11

bachp committed:
unifiStable: 5.12.35 -> 5.12.66

danielfullmer committed:
wifite2: 2.2.5 -> 2.5.2

The kimocoder repo is more recently updated, and is the one used by
Debian / Kali Linux as well.

AndersonTorres committed:
atari++: 1.81 -> 1.83

jonringer committed:
python3Packages.ibis: 1.2.0 -> 1.3.0, fix tests

Commit:
ursadb: init at v1.2.0

r-ryantm committed:
kubernetes: 1.17.3 -> 1.18.0

worldofpeace committed:
calibre: install desktop items

These have been available for a while it seems.

jluttine committed:
xonotic: use svg icons

prikhi committed:
lightdm-mini-greeter: 0.3.4 -> 0.4.0

prikhi committed:
lightdm-mini-greeter: Add Changelog Link

Ma27 committed:
nixos/systemd-nspawn: disallow multiple packages with `.nspawn`-units

In contrast to `.service`-units, it's not possible to declare an
`overrides.conf`, however this is done by `generateUnits` for `.nspawn`
units as well. This change breaks the build if you have two derivations
configuring one nspawn unit.

This will happen in a case like this:

``` nix
{ pkgs, ... }: {
  systemd.packages = [
    (pkgs.writeTextDir "etc/systemd/nspawn/container0.nspawn" ''
      [Files]
      Bind=/tmp
    '')
  ];
  systemd.nspawn.container0 = {
    /* ... */
  };
}
```

AndersonTorres committed:
cimg: 2.8.4 -> 2.9.0

romildo committed:
xfce: add pos attribute in mkXfcederivation

AndersonTorres committed:
sound-of-sorting: 2015-07-21 -> 2017-12-23

dominikh committed:
luminance-hdr: use Qt5's mkDerivation

AndersonTorres committed:
cherrytree: 0.39.0 -> 0.39.2

AndersonTorres committed:
higan: 106 -> 110

romildo committed:
xfce.orage: the git repository has been archived (unmaintained)

AndersonTorres committed:
icewm: update URL

TravisWhitaker committed:
LLVM: Always pass host/target info to compiler-rt/libstdcxxClang

AndersonTorres committed:
jupp: 31 -> 39

AndersonTorres committed:
mksh: 57 -> 58

AndersonTorres committed:
rs: init at 20200313

gnprice committed:
treewide: Fix up stripDebugList attrs to be lists.

The documentation says this should be a list, and it already is in
about half the expressions that set it.

The difference doesn't matter at present, because these values are all
space-free literals.  But it will in a future with __structuredAttrs .

(The similar attr stripAllList has no users in the nixpkgs tree, so
there's nothing to do to fix any of those up.)

gnprice committed:
doc/stdenv: Update default for stripDebugList.

This has changed a few times from 2008 to 2012: cdc74e167, d9213df2c,
and 095db9fe3.

AndersonTorres committed:
ppsspp: 1.9.3 -> 1.9.4

AndersonTorres committed:
gp2c: 0.0.11pl2 -> 0.0.11pl3

AndersonTorres committed:
libtap: 1.12.0 -> 1.14.0

vbgl committed:
ocamlPackages.duration: init at 0.1.3

vbgl committed:
ocamlPackages.dns: init at 4.4.1

vbgl committed:
ocamlPackages.randomconv: init at 0.1.3

vbgl committed:
ocamlPackages.mirage-clock: init at 3.0.1

vbgl committed:
ocamlPackages.mirage-device: init at 2.0.0

vbgl committed:
ocamlPackages.mirage-flow: init at 2.0.1

vbgl committed:
ocamlPackages.mirage-protocols: init at 4.0.1

vbgl committed:
ocamlPackages.mirage-random: init at 2.0.0

vbgl committed:
ocamlPackages.mirage-stack: init at 2.0.1

vbgl committed:
ocamlPackages.dns-client: init at 4.4.1

prusnak committed:
jitsi-meet-electron: init at 2.0.0

oxalica committed:
rust-analyzer: change to nix-prefetch and fix version checking in update script

oxalica committed:
rust-analyzer: 2020-03-16 -> 2020-04-06

oxalica committed:
vscode-extensions.rust-analyzer: 2020-03-16 -> 2020-04-06

bhipple committed:
ethminer: mark as broken

Doesn't build with gcc9, and if overlayed to use gcc8 stdenv fails on CUDA issues.

flokli committed:
subsurface: 4.8.2 -> 4.9.3

This bumps subsurface to 4.9.3.

As the subsurface-specific libdivecomputer source is shipped as a git
submodule these days, we can avoid having to fetch a separate tarball
that needs to be kept in sync (and is only uploaded for each release).

This also applies a patch on top of 4.9.3 to properly handle libgit2
versions 1.0 and later - conditional code for older libgit versions has
been removed upstream, but that's a nontrivial backport, which is why
this patch was added.

cole-h committed:
qimgv: enable video support

I actually went back and `strace`'d the binary when it complained about
being unable to load the library it uses for playing videos. Turns out,
it wasn't finding the library because it wasn't in any of its library
paths. I added the lib path to `LD_LIBRARY_PATH` so it can find the
library it uses to play videos, and now things are peachy.

There is a (seemingly innocuous) error that gets displayed from Exiv2
being unable to determine its image type. Since it's actually a video, I
think it's fine. Another issue that pops up in the output is missing
`libcuda.so.1`, but that doesn't seem to affect functionality, and using
`addOpenGLRunpath` on both the binary and libraries didn't silence it.

Also did a little formatting.

worldofpeace committed:
bug_report.md: ask to notify maintainers

This is really essential thing that everyone should be doing.
A few extra notifications won't hurt anyone, and we should be tracking issues
to connect them to the best people who can handle them.

Co-Authored-By: Cole Helbling <cole.e.helbling@outlook.com>

oxalica committed:
vscode-extensions.rust-analyzer: use global NPM packages to share dependencies

oxalica committed:
nodePackages: update

prusnak committed:
Revert "riot-desktop: wrap with wrapGAppsHook"

This reverts commit fe6addbbf717d4977434e81acd7535c3b074e3a3.

prusnak committed:
electron: fix wrapGAppsHook usage

marsam committed:
victoriametrics: 1.34.5 -> 1.34.7

Profpatsch committed:
skalibs: 2.9.2.0 -> 2.9.2.1

r-ryantm committed:
offlineimap: 7.3.2 -> 7.3.3

iblech committed:
iodine: improve test in view of #58806

snicket2100 committed:
advancecomp: version number correctly embedded in binaries

without this fix, binaries report their version number as "vnone"
which for example stops `image_optim` from using `advpng` (refuses
to use it not being able to tell what version number it has)

minijackson committed:
jellyfin: 10.5.3 -> 10.5.4

thoughtpolice committed:
liburing: 0.4pre514_2454d630 -> 0.6pre600

Signed-off-by: Austin Seipp <aseipp@pobox.com>

minijackson committed:
jellyfin: add link to nixos test

sternenseemann committed:
ocamlPackages.eqaf: 0.5 -> 0.6

sternenseemann committed:
ocamlPackages.eqaf: enable cstruct and bigarray modules

sternenseemann committed:
mirage-crypto{,-rng,-pk}: init at 0.6.2

sternenseemann committed:
ocamlPackages.mirage-clock-unix: init at 3.0.1

flokli committed:
nixos/networking: fix setting .macAddress and .mtu with networkd

This needs to be set in the .linkConfig of a .network

flokli committed:
nixos/networking: add assertion catching setting mac addresses on tun devices

Setting a MAC Address on a tun interface isn't supported, and invoking
the corresponding command fails.

flokli committed:
nixos/networking: move network-link-${i.name} to scripted networking

The unit sets MTU and MAC Address even with networkd enabled, which
isn't necessary anymore, as networkd handles this by itself.

sternenseemann committed:
ocamlPackages.mirage-flow-unix/-combinators: init at 2.0.1

sternenseemann committed:
ocamlPackages.fmt: 0.8.6 -> 0.8.8

sternenseemann committed:
ocamlPackages.ounit2: init at 2.2.2

sternenseemann committed:
ocamlPackages.ocaml-syntax-shims: init at 1.0.0

sternenseemann committed:
ocamlPackages.index: 1.0.1 -> 1.2.0

Note: tests are available, should be added in the future (crowbar is required)

sternenseemann committed:
ocamlPackages.irmin*: 2.0.0 -> 2.1.0

updated:

* irmin
* irmin-fs
* irmin-git
* irmin-graphql
* irmin-http
* irmin-mem
* irmin-pack (also added new dependecy ocaml_lwt)
* irmin-test
* irmin-unix
* irmin-watcher

sternenseemann committed:
ocamlPackages.ppx_irmin: init at 2.1.0

Ma27 committed:
nixos/build-vms: propagate file location

When trying to build a VM using `nixos-build-vms` with a configuration
that doesn't evaluate, an error "at `<unknown-file>`" is usually shown.

This happens since the `build-vms.nix` creates a VM-network of
NixOS-configurations that are attr-sets or functions and don't contain
any file information. This patch manually adds the `_file`-attribute to
tell the module-system which file contained broken configuration:

```
$ cat vm.nix
{ vm.invalid-option = 1; }

$ nixos-build-vms vm.nix
error: The option `invalid-option' defined in `/home/ma27/Projects/nixpkgs/vm.nix@node-vm' does not exist.
(use '--show-trace' to show detailed location information)
```

globin committed:
alertmanager: implement HA clustering support

flokli committed:
nixosTests.networking.virtual: fix with networkd

We only need to wait for network.target to get up, and the
network-addresses-${interfaceName} units are scripted networking only.

flokli committed:
nixosTests.networking: make routing table comparison more reliable

This was whitespace-sensitive, kept fighting with my editor and broke
the tests easily. To fix this, let python convert the output to
individual lines, and strip whitespace from them before comparing.

flokli committed:
nixosTests.networking: test setting MTU and MAC Address

Both the scripted and networkd backend now support setting MTU and MAC
Address, so do this in a test to ensure it doesn't break.

bbigras committed:
zenith: 0.8.1 -> 0.8.2

globin committed:
sslscan: enabling scanning for sslv3

jtojnar committed:
inkscape: format with nixpkgs-fmt

Also order the build inputs alphabetically and switch to pname+version.

r-ryantm committed:
python37Packages.sphinxcontrib-katex: 0.5.1 -> 0.6.0

romildo committed:
common-updater-scripts: add scripts to help update packages

- updateScript:
  A nix expression that can be used in passThrough to update a package

- list-git-tags:
  A shell script to list available tags in a git repository

- list-archive-two-level-versions:

  A shell script to list available versions in a web site in two
  levels: there is a page listing the available major.minor versions,
  and for each of them there is another page listings the patch level
  versions major.minor.patch.

  It is suitable for Xfce packages for instance.

How the updater works:

1. collect the available versions from the source repository (using a
script given as argument)

2. print the collected versions (for debugging)

3. (optionally) apply some transformation to the collected versions to
make them compatible with the versions used in nixpkgs (for instance,
tags in the Xfce git repository may be prefixed with the package name,
and the prefix need to be removed)

4. sort the available versions in decreasing order

5. choose the first version that pass validation:
   - check if the version may be a development version

   - if the version IS unstable, skip it and give a warning about
   skipping a development version (for debugging)

   - if the version COULD BE unstable, take it and give a warning
   about taking a potential development version (for debugging)

   - if the version IS stable, take it

6. update the package version and checksum in its nix expression

7. print the git commands for adding the modified files and for
committing the changes

romildo committed:
xfce: tailor general update scripts to the xfce packages

romildo committed:
xfce: add update scripts to xfce packages

Br1ght0ne committed:
ulauncher: 5.6.1 -> 5.7.3

veprbl committed:
wpsoffice: 11.1.0.9080 -> 11.1.0.9505

romildo committed:
common-updater-scripts: ignore update-git-commits.txt

It is generated by common-updater/update-script.nix

r-ryantm committed:
python27Packages.swagger-ui-bundle: 0.0.5 -> 0.0.6

m1cr0man committed:
acme: share accounts between certificates

There are strict rate limits on account creation for Let's Encrypt
certificates. It is important to reuse credentails when possible.

jtojnar committed:
inkscape: 0.92.4 → 0.92.5

https://gitlab.com/inkscape/inkscape/-/tags/INKSCAPE_0_92_5

jtojnar committed:
inkscape: add jtojnar as maintainer

jtojnar committed:
inkscape: clean up

The unpackPhase override was only needed when the download server mangled the file extension:

https://github.com/NixOS/nixpkgs/pull/23652

Commit:
mame: 0.219 -> 0.220 (#84978)

jtojnar committed:
lib/options: Relax showOption quoting

https://github.com/NixOS/nixpkgs/commit/124cccbe3b63122733e02e41e45a383ec48752fd
broke the build of NixOS manual.

It does not make sense to be as strict as with attributes since we
are not limited by the CLI's inability to handle numbers.
Placeholders should not be quoted either as they are not part of Nix
syntax but a meta-level construct.

xrelkd committed:
tokei: 11.0.0 -> 11.1.0

zowoq committed:
miniserve: build on darwin (#85187)

rakesh-formbay committed:
pythonPackages.py-multibase: init at 1.0.1

veprbl committed:
wpsoffice: apply nixpkgs-fmt, sort alphabetically

r-ryantm committed:
blugon: 1.12.0 -> 1.12.1

veprbl committed:
wpsoffice: switch to autoPatchelfHook, update dependencies

veprbl committed:
wpsoffice: apply Qt and GTK wrappers

Based on the work of Ninlives.

r-ryantm committed:
carla: 2.0.0 -> 2.1

veprbl committed:
wpsoffice: remove outdated substituteInPlace

basilgood committed:
tig: fix tig-completion's dependency on __git_complete

tig recently updated it's bash-completion making it depend on __git-complete from git.
Becase __git-complete is not automatically sourced tig bash completion fails.
Also this PR makes tig completion load on-demand.

Commit:
starship: 0.40.0 -> 0.40.1

saschagrunert committed:
cri-o: 1.17.1 -> 1.17.3

Signed-off-by: Sascha Grunert <sgrunert@suse.com>

adisbladis committed:
kdeApplications.kbreakout: init at 19.12.3

adisbladis committed:
kdeApplications.kspaceduel: init at 19.12.3

adisbladis committed:
crack-attack: Set correct game directories

adisbladis committed:
pokerth: Add Qt wrapper

adisbladis committed:
xmoto: Fix missing text

adisbladis committed:
mpv: Enable archive support by default

adisbladis committed:
firefox-beta-bin: 76.0b1 -> 76.0b4

adisbladis committed:
firefox-devedition-bin: 76.0b1 -> 76.0b4

sternenseemann committed:
ocamlPackages.notty: 0.2.1 → 0.2.2

marsam committed:
netdata: 1.21.0 -> 1.21.1

Changelog: https://github.com/netdata/netdata/releases/tag/v1.21.1

marsam committed:
gitAndTools.git-trim: 0.2.4 -> 0.3.1

marsam committed:
netdata-go.d.plugin: 0.15.0 -> 0.18.0

jtojnar committed:
inkscape: drop redundant enableParallelBuilding

With CMake it is on by default.

prusnak committed:
gtk-mac-integration: 2.0.8 -> 2.1.3

veprbl committed:
wpsoffice: drop i686-linux from platforms

markuskowa committed:
globalarrays: 5.7 -> 5.7.1

svanderburg committed:
dysnomia: 0.9 -> 0.9.1

svanderburg committed:
nixos/dysnomia: fix documentRoot property

lheckemann committed:
nixos/libinput: refer to libinput manual

etu committed:
php.packages.phpcbf: 3.5.3 -> 3.5.4

Changelog:
https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.5.4

Also switch to phpWithExtensions because it complained about missing
extensions when running the binary.

etu committed:
php.packages.phpcs: 3.5.3 -> 3.5.4

Changelog:
https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.5.4

Also switch to phpWithExtensions because it complained about missing
extensions when running the binary.

etu committed:
php.packages.phpstan: 0.12.14 -> 0.12.18

Changelog: https://github.com/phpstan/phpstan/releases/tag/0.12.18

Also switch to phpWithExtensions because it complained about missing
extensions when running the binary.

leenaars committed:
Sylk: 2.5.0 -> 2.6.1 (#85213)

etu committed:
testssl: 3.0rc6 -> 3.0

Changelog: https://github.com/drwetter/testssl.sh/releases/tag/3.0

etu committed:
sequeler: 0.7.4 -> 0.7.9

Changelogs:
 - https://github.com/Alecaddd/sequeler/releases/tag/v0.7.5
 - https://github.com/Alecaddd/sequeler/releases/tag/v0.7.6
 - https://github.com/Alecaddd/sequeler/releases/tag/v0.7.7
 - https://github.com/Alecaddd/sequeler/releases/tag/v0.7.8
 - https://github.com/Alecaddd/sequeler/releases/tag/v0.7.9

romildo committed:
humanity-icon-theme: init at 0.6.15

AndersonTorres committed:
biblesync: 1.1.2 -> 2.0.1

AndersonTorres committed:
sword: 1.7.4 -> 1.8.1

mdlayher committed:
corerad: 0.2.2 -> 0.2.3

nomeata committed:
wasmtime: Do not prefix version with `v`

it seems that this breaks the use of `nix-env -i wasmtime`

LnL7 committed:
Revert "lib/options: Use escapeNixIdentifier for showOption"

This is used in in the manual generation for option identifiers that can
be linked. This, unike what the example describes, doesn't preserve
quotes which is needed for these identifiers to be valid.

This reverts commit 124cccbe3b63122733e02e41e45a383ec48752fd.

LnL7 committed:
lib/options: fix showOption example

danielfullmer committed:
wifite2: add danielfullmer to maintainers

sternenseemann committed:
ocamlPackages.asn1-combinators: 0.2.0 -> 0.2.2

primeos committed:
wf-recorder: 0.2 -> 0.2.1

Commit:
openttd: 1.10.0 -> 1.10.1

1.10.0 added SDL support

primeos committed:
lf: Install the shell completions

flokli committed:
nixosTests.flannel: port to python, unbreak

For reasons yet unknown, the vxlan backend doesn't work (at least inside
the qemu networking), so this is moved to the udp backend.

Note changing the backend apparently also changes the interface name,
it's now `flannel0`, not `flannel.1`

fixes #74941

marsam committed:
nodejs-13_x: 13.12.0 -> 13.13.0

Changelog: https://github.com/nodejs/node/releases/tag/v13.13.0

veprbl committed:
wpsoffice: remove specific output names

ryantm committed:
linode-cli: remove deprecated perl version, init python version at 2.14.1

r-ryantm committed:
notejot: 1.6.0 -> 1.6.3

peterhoeg committed:
freerdp: re-enable tests except for one failing test

r-ryantm committed:
otfcc: 0.9.6 -> 0.10.4

Commit:
git: Allow the update script to target non-latest versions

This came in handy when I wanted to bump a patch version while avoiding
a new minor version.

r-ryantm committed:
pgmetrics: 1.8.1 -> 1.9.0

r-ryantm committed:
rdma-core: 28.0 -> 29.0

r-ryantm committed:
smarty3: 3.1.34 -> 3.1.35

marsam committed:
fira-code: 3 -> 3.1

Changelog; https://github.com/tonsky/FiraCode/releases/tag/3.1

r-ryantm committed:
nix-zsh-completions: 0.4.3 -> 0.4.4 (#85267)

andersk committed:
system-syzygy: Fix invalid desktopFile arguments

Fixes this build failure:

Running desktop-file validation
/nix/store/vadj03624kv6h963c7fjwdz1hg403aa9-system-syzygy.desktop/share/applications/system-syzygy.desktop: error: value "%out%/bin/syzygy" for key "Exec" in group "Desktop Entry" contains an invalid field code "%o"
/nix/store/vadj03624kv6h963c7fjwdz1hg403aa9-system-syzygy.desktop/share/applications/system-syzygy.desktop: error: value "%out%/bin/syzygy" for key "Exec" in group "Desktop Entry" contains an invalid field code "%/"

Signed-off-by: Anders Kaseorg <andersk@mit.edu>

r-ryantm committed:
zafiro-icons: 1.0 -> 1.1

r-ryantm committed:
acpica-tools: 20200110 -> 20200326

r-ryantm committed:
atlassian-cli: 9.1.1 -> 9.2.0

r-ryantm committed:
bibclean: 3.03 -> 3.04

r-ryantm committed:
bitlbee-discord: 0.4.2 -> 0.4.3

NeQuissimus committed:
zoom-us: 3.5.383291.0407 -> 3.5.385850.0413

r-ryantm committed:
python27Packages.autopep8: 1.5 -> 1.5.1

das-g committed:
chromedriver: 80.0.3987.16 -> 81.0.4044.69

romildo committed:
common-updater: add update script to all-packages

romildo committed:
xfce: use updateScript from pkgs

r-ryantm committed:
chirp: 20200213 -> 20200409

r-ryantm committed:
cfdg: 3.2_2 -> 3.3

vbgl committed:
Revert "[Don’t merge] Recurse into OCaml packages"

This reverts commit e16c7dae256bcb6b77399a4d80f512a117cca5d0.

r-ryantm committed:
cjdns: 20.5 -> 20.6

Ma27 committed:
iftop: add `passthru.tests` to reference VM-test

Ma27 committed:
grocy: add `passthru.tests` to reference VM-test

Ma27 committed:
cargo-make: 0.30.4 -> 0.30.5

https://github.com/sagiegurari/cargo-make/releases/tag/0.30.5

Ma27 committed:
documize-community: add `passthru.tests` to reference VM-test

Ma27 committed:
nextcloud*: add `passthru.tests` to reference VM-tests

worldofpeace committed:
ping: use vala_0_40

fails to build otherwise.

worldofpeace committed:
nasc: fix build

zowoq committed:
podman: 1.8.2 -> 1.9.0

https://github.com/containers/libpod/releases/tag/v1.9.0

lopsided98 committed:
g2o: fix log limit exceeded error in Hydra

lopsided98 committed:
g2o: unstable-2019-04-07 -> 20200410

xrelkd committed:
eksctl: 0.16.0 -> 0.17.0

worldofpeace committed:
agenda: 1.0.12 -> 1.1.0

https://github.com/dahenson/agenda/releases/tag/1.1.0

worldofpeace committed:
monitor: 0.6.2 -> 0.7.1

https://github.com/stsdc/monitor/releases/tag/0.7.0
https://github.com/stsdc/monitor/compare/0.7.0...0.7.1

worldofpeace committed:
monitor: add note how to use indicator.

worldofpeace committed:
ephemeral: 6.3.1 -> 6.3.3

https://github.com/cassidyjames/ephemeral/releases/tag/6.3.2
https://github.com/cassidyjames/ephemeral/releases/tag/6.3.3

worldofpeace committed:
pantheon.elementary-videos: 2.7.0 -> 2.7.1

https://github.com/elementary/videos/releases/tag/2.7.1

worldofpeace committed:
pantheon.pantheon-agent-geoclue2: 1.0.3 -> 1.0.4

https://github.com/elementary/pantheon-agent-geoclue2/releases/tag/1.0.4

zowoq committed:
fd: 7.5.0 -> 8.0.0

https://github.com/sharkdp/fd/releases/tag/v8.0.0

Ma27 committed:
nixos/doc: fix database-setup example for matrix-synapse

Closes #85327

turion commented:
@cdepillabout didn't you mean _stackage_, not hackage?

cdepillabout commented:
@jappeace What are you referring to?

`cut-the-crap` was originally not uploaded to Hackage, so it didn't automatically appear in the Haskell package set here in nixpkgs.

However, it appears that @jappeace has since uploaded to Hackage, so now it should be easy to just use `haskellPackages.cut-the-crap`.

Although you're right that uploading to Stackage (in addition to Hackage) would make it easier to make sure `cut-the-crap` builds here in nixpkgs.

primeos committed:
google-chrome-{beta,dev}: Fix one substituteInPlace pattern

turion commented:
@cdepillabout Thanks for the clarification! I can never remember how, in which files, and in which workflow either hackage or stackage are crawled. (Wish this was documented.)

turion commented:
@cdepillabout Thanks for the clarification! I can never remember how, in which files, and in which workflow either hackage or stackage are crawled. (Wish this was documented.)

cdepillabout commented:
@turion Yeah, I definitely sympathize!

A (very short) explanation is that Haskell packages are updated from Hackage automatically on a daily basis in this file:

https://github.com/NixOS/nixpkgs/blob/16477d764f375ef7b81acbfd13189299ff67f233/pkgs/development/haskell-modules/hackage-packages.nix

The packages in the stackage resolver are specified here:

https://github.com/NixOS/nixpkgs/blob/16477d764f375ef7b81acbfd13189299ff67f233/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L45-L80

This list is updated once a week or so by peti.  I believe he does it semi-automatically.

That file also has a bunch of other settings for special things to do when updating packages from Hackage.

jappeace commented:
I'll fix this up right now as discussed above

jappeace committed:
cut-the-crap: init at 1.0.0

jappeace commented:
Ok nice, that's a lot less code and it makes updating this as simple as re-uploading to hackage. Thanks for all the help @cdepillabout

cdepillabout commented:
@jappeace One more thing.  Could you rebase this on the `haskell-updates` branch, and change the base branch for this PR here on GitHub to `haskell-updates`?

We like to have most Haskell-related changes going to the `haskell-updates` branch.

jappeace commented:
lol, that upsets github quite a bit @cdepillabout

jappeace commented:
I don't want to deal with github bullshit, I kindly refer you to https://github.com/NixOS/nixpkgs/pull/85438

Sorry for everyone else who got tagged. I didn't expect github to be so un-intelligent.

doronbehar commented:
@jappeace please mind your language :pray: . It's 1 thing we add here a program that has "crap" in it's name - it doesn't mean we have to use such words when we deal with them.

jappeace commented:
@doronbehar I already self censored. You probably had a tab open or something.

doronbehar commented:
I noticed that and I still felt it was necessary to put it. I was subscribed to this issue and I got that by email.
